### PR TITLE
[fix]:[pie-6386]: fix repo filter for deployments page

### PR DESCRIPTION
**Pull request by adivishy1**
## Describe your changes
Code to fix repo filter only for deployments page. Have tested the out following:
1. Only repo filter in the deployments page - new GitX
2. Without repo filter in deployments page - new GitX
3. Repo and branch filter in deployments page - new GitX
4. Execution page - new GitX
5. Execution history page with branch filter - GitX
6. Deployments page - old git sync
7. Execution history page - old git sync

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/40232)

### CODE REVIEW

**Review by mohitg0795 — APPROVED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40232)* status: :shipit: complete! ✓

File: pipeline-service/service/src/main/java/io/harness/pms/plan/execution/service/PMSExecutionServiceImpl.java [hunk 1]
```diff
@@ -168,22 +169,18 @@ public Criteria formCriteria(String accountId, String orgId, String projectId, S
...
 
     Criteria gitCriteria = new Criteria();
-    if (gitSyncBranchContext != null) {
-      Criteria gitCriteriaDeprecated =
-          Criteria.where(PlanExecutionSummaryKeys.gitSyncBranchContext).is(gitSyncBranchContext);
-
-      EntityGitDetails entityGitDetails = pmsGitSyncHelper.getEntityGitDetailsFromBytes(gitSyncBranchContext);
-      Criteria gitCriteriaNew =
-          Criteria.where(PlanExecutionSummaryKeys.entityGitDetailsBranch).is(entityGitDetails.getBranch());
-      if (entityGitDetails.getRepoIdentifier() != null
```
[the remark this answers is not in the document]
Comment (adivishy1) [reply pipeline-service/service/src/main/java/io/harness/pms/plan/execution/service/PMSExecutionServiceImpl.java:178]: added it back

**Review by archit-harness — DISMISSED** [review]
> Reviewed 3 of 7 files at r2, 2 of 2 files at r3, 3 of 3 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40232)* status: :shipit: complete! ✓

**Review by archit-harness — APPROVED** [review]
> Reviewed 3 of 3 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40232)* status: :shipit: complete! ✓

**Review by namanvermah — APPROVED** [review]
> Reviewed 2 of 7 files at r2, 1 of 2 files at r3, 2 of 3 files at r4, 3 of 3 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40232)* status: :shipit: complete! ✓

**Review by archit-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40232)* status: :shipit: complete! ✓

**Pull request merged**